src/transformers/quantizers/quantizer_hqq.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from .base import HfQuantizer
+
+
+if TYPE_CHECKING:
+    from ..modeling_utils import PreTrainedModel
+
+from ..integrations import prepare_for_hqq_linear
+from ..utils import is_accelerate_available, is_hqq_available, is_torch_available, logging
+from ..utils.hqq_utils import find_parent
+from .quantizers_utils import get_module_from_name
+
+
+if is_accelerate_available():
+    from accelerate.hooks import remove_hook_from_module
+
+if is_torch_available():
+    import torch
+
+logger = logging.get_logger(__name__)
+
+
+class HQQHfQuantizer(HfQuantizer):
+    """
+    HQQ quantizer base HF class.
+    nn.Linear modules are first tagged with quant_config in _process_model_before_weight_loading().
+    The actually quantization and offloading to the GPU is done in check_quantized_param().
+    self.show_progress (bool) is used to show quantization progress in each shard.
+    """
+
+    use_keep_in_fp32_modules = False
+    requires_parameters_quantization = True
+    requires_calibration = False
+    required_packages = ["hqq"]
+
+    def __init__(self, quantization_config, **kwargs):
+        super().__init__(quantization_config, **kwargs)
+        self.show_progress = quantization_config.show_progress
+        self.torch_dtype = None
+        self.using_multi_gpu = False
+
+    def validate_environment(self, *args, **kwargs):
+        if not (is_hqq_available()):
+            raise ImportError(
+                "HQQ is not available. Please follow the instructions to install it: `https://github.com/mobiusml/hqq/`"
+            )
+
+        if kwargs.get("from_tf", False) or kwargs.get("from_flax", False):
+            raise ValueError(
+                "Converting weights from tf/flax weights is currently not supported, please make"
+                " sure the weights are in PyTorch format."
+            )
+
+        if not torch.cuda.is_available():
+            raise RuntimeError("No GPU found. A GPU is needed for quantization.")
+
+        if self.torch_dtype is None:
+            self.torch_dtype = kwargs.get("torch_dtype", torch.float16)
+
+        if self.using_multi_gpu is False:
+            if "device_map" in kwargs:
+                if isinstance(kwargs["device_map"], dict):
+                    self.using_multi_gpu = len({item[1] for item in kwargs["device_map"].items()}) > 1
+
+    def check_quantized_param(
+        self,
+        model: "PreTrainedModel",
+        param_value: "torch.Tensor",
+        param_name: str,
+        state_dict: Dict[str, Any],
+        **kwargs,
+    ) -> bool:
+        module, tensor_name = get_module_from_name(model, param_name)
+
+        if isinstance(module, torch.nn.Linear):
+            return True
+        else:
+            return False
```

Comment:
```suggestion
        returnif type(module) is torch.nn.Linear
```

Reply:
It breaks with this change, is another one possible?

Reply:
https://github.com/huggingface/transformers/pull/29637/commits/3a5679a9eb5d093246d00d6edd8265409900ec4e

Reply:
Updated https://github.com/huggingface/transformers/pull/29637/commits/c7ec12399b6fd6fedf828ea62b66fb77e969c03c